rfcs/0052-create-sink-into-table.md

```diff
@@ -0,0 +1,400 @@
+---
+feature: create_sink_into_table
+authors:
+  - "st1page"
+  - "Eric Fu"
+start_date: "2023/02/15"
+---
+
+# Create Sink into Table
+
+## Background && Summary
+
+Currently, RisingWave use two ways to create a streaming job. They both construct a streaming job to maintain the changes of the streaming query's result continually. But it 
+- `CREATE MATERIALIZED VIEW`.
+  - Materialize the result as a RW's table and apply the changes on it.
+  - The table is coupled with the streaming job. They trust some property of each other.
+    - The streaming job is the only changes source of the table. No other changes such as DML is allowed on this table.
+    - The table's PK and other constraint is inferred by the streaming query's own property. They trust each other and do not need additional check.
+    - The result table is in the same global checkpoint with the upstream table. So the changes are applied exactly-once naturally. 
+- `CREATE SINK`
+  - Apply the changes into an external system's table(such as MySQL).
+  - The table in the external storage and the streaming job in RW do not trust any assumption each other.
+    - The user can do DML on the external in the storage. The can create multiple sink into the same table too.
+    - The table's PK and other constraint is maintained by the external storage. 
+    - The RW's streaming execution need additional 2pc mechanism with external storage to achieve exactly-once.
+
+This RFC give a new SQL grammar `CREATE SINK sink_name TO table_name AS SELECT ...` to create streaming job and apply its changes into a RW's table, which decouples the streaming job and the result table.
+- It's changes events behavior is same with other external sinks.
+- When table's constraint is violated by the changes, the behavior should be determined by the RW's table. Specifically, it can be defined by ON CONFLICT clause ([RFC: on conflict clause](https://github.com/risingwavelabs/rfcs/pull/48)). 
+- align the barrier and epoch with upstream, satisfying exactly-once and snapshot isolation with upstream.
+
+Also, the `ALTER MATERIALIZED VIEW [mv_table] TO TABLE` is used for transforming an existent MV to a table and a Sink to the table. 
+## Motivation
+There are some use cases.
```

Comment:
Should we add these use cases to the user doc? https://docs.risingwave.com/docs/current/sql-create-sink-into/

Currently, nothing is mentioned there. I doubt whether users can understand these abilities.

Reply:
Oh, I see https://github.com/risingwavelabs/risingwave-docs/issues/2592 mentioned guides are missing